src/ucp/wireup/select.c

```diff
@@ -1836,7 +1836,9 @@ ucp_wireup_add_rma_bw_lanes(const ucp_wireup_select_params_t *select_params,
     ucp_wireup_init_select_flags(&peer_rma_flags, 0, 0);
 
     if ((ep_init_flags & UCP_EP_INIT_CREATE_AM_LANE_ONLY) ||
-        (context->config.ext.max_rndv_lanes == 0)) {
+        (context->config.ext.max_rndv_lanes == 0) ||
+        (context->config.ext.rndv_mode == UCP_RNDV_MODE_AM) ||
+        (context->config.ext.rndv_mode == UCP_RNDV_MODE_RKEY_PTR)) {
```

Comment:
- we also use rma_bw lanes for RMA
- in theory we can fail to use one of the mentioned rndv schemes and fallback to some other which is using rma_bw lanes

Reply:
agree with the first point (so we need to check the feature flags),
regarding the second point - the fallback is always to AM scheme that does not use RMA lanes

Reply:
relying on rndv_scheme still seems risky/error prone to me. Recently we changed the fallback flow for rndv protocols when requested scheme is not available, so it can happen again in the future and this code can implicitly affect that